src/components/metadata/metadata.test.js

```diff
@@ -93,70 +97,59 @@ describe('MetaData', () => {
         true
       );
     });
 
     it('shows the node name as the title', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       expect(textOf(title(wrapper))).toEqual(['salmon']);
     });
 
     it('shows the node type as text', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       const row = rowByLabel(wrapper, 'Type:');
       expect(textOf(rowValue(row))).toEqual(['task']);
     });
 
     it('does not display the node parameter row when there are no parameters', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
       const row = rowByLabel(wrapper, 'Parameters:');
-      //this is output of react-json-view with no value
+      //this is the metadata output when there is no data
       expect(textOf(rowObject(row))).toEqual(['-']);
     });
 
     it('shows the node parameters when there are parameters', () => {
-      const metadata = getClickedNodeMetaData(
-        prepareState({
-          data: animals,
-          afterLayoutActions: [
-            () => toggleNodeClicked(salmonTaskNodeId),
-            () =>
-              addNodeMetadata({ id: salmonTaskNodeId, data: node_parameters }),
-          ],
-        })
-      );
-      const wrapper = setup.mount(
-        <MetaData visible={true} metadata={metadata} />
-      );
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_parameters,
+      });
       const row = rowByLabel(wrapper, 'Parameters:');
       //this is output of react-json-view with 3 parameters
       expect(textOf(rowObject(row))[0]).toEqual(
         expect.stringContaining('3 items')
       );
     });
 
-    it('shows the node inputs when parameters are disabled (default)', () => {
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
-      const row = rowByLabel(wrapper, 'Inputs:');
-      expect(textOf(rowValue(row))).toEqual(['Cat', 'Dog']);
-    });
-
-    it('shows the node inputs when parameters are enabled', () => {
-      const mount = (props) => {
-        return setup.mount(<MetaData visible={true} />, {
-          beforeLayoutActions: [() => toggleTypeDisabled('parameters', false)],
-          afterLayoutActions: [
-            () => {
-              // Click the expected node
-              return toggleNodeClicked(props.nodeId);
-            },
-          ],
-        });
-      };
-      const wrapper = mount({ nodeId: salmonTaskNodeId });
+    it('shows the node inputs', () => {
+      const wrapper = mount({
+        nodeId: salmonTaskNodeId,
+        mockMetadata: node_task,
+      });
```

Comment:
To test the node inputs, I would now reformat the test as `shows the node inputs with metadata fetched from the endpoint`, and then reference the setup of the `toggle code` test to format this test. ( this is the same setup given the code toggle is also relying on returned metadata to format the test.)

```describe('shows the node inputs with metadata fetched from the endpoint', () => {
      // Get metadata for a sample node
      const metadata = getClickedNodeMetaData(
        prepareState({
          data: animals,
          afterLayoutActions: [() => toggleNodeClicked(salmonTaskNodeId)],
        })
      );

      const wrapper = setup.mount(
        <MetaData visible={true} metadata={metadata} />
      );

      lett row = rowByLabel(wrapper, 'Inputs:');
      expect(textOf(rowValue(row))).toEqual([
        'Cat',
        'Dog',
        'Parameters',
        'Params:rabbit',
      ]);
    });

      row = rowByLabel(wrapper, 'Outputs:');
      expect(textOf(rowValue(row))).toEqual(['Horse', 'Sheep']);

    });```